### PR TITLE
1998 h1s with attribute markings

### LINKED ISSUE #1998: H1's in the 'single title' assessment are not marked when it has html-attributes.
Marking H1's using the eye marker next to the single title assessment will not mark H1's with attributes in the opening tag.

Parent #1903

### PR DESCRIPTION
**Pull request by hansjovis**
## Summary

**Note**: Please review and test #1997 before this one, since this PR is based on changes made in that PR.

This PR can be summarized in the following changelog entry:

* Improves the markings of the single title assessment. It now marks H1-headers with attributes in the opening tag as well.

## Relevant technical choices:

* Removed the `tag: "h1"` from the H1's found by the `h1s` researcher since the function will only return H1's.

## Test instructions

This PR can be tested by following these steps:

* Start up the devtool in recalibration mode.
* Enter a random (html) text.
* Add two or more H1's, one at the start and the other ones in the middle of the text.
* Press the eye mark button next to the single title assessment in the assessment results.
* Scroll down to the markings paper.
* [ ] The H1's in the middle should be marked, the one at the start should not be. NB: Use different title text!
* Add some attributes to one or more H1's in the middle, like `id='some_header' class='some classes'`.
* [ ] The H1's with attributes should still be marked.

Fixes #1998 
Requires #1964

### CODE REVIEW

File: src/assessments/seo/SingleH1Assessment.js [hunk 1]
```diff
@@ -121,9 +121,10 @@ class singleH1Assessment extends Assessment {
 		}
 
 		return map( h1s, function( h1 ) {
+			const attributes = h1.attributes ? h1.attributes : "";
 			return new Mark( {
-				original: "<h1>" + h1.content + "</h1>",
-				marked: "<h1>" + marker( h1.content ) + "</h1>",
+				original: `<h1${attributes}>${h1.content}</h1>`,
+				marked: `<h1${attributes}>${marker( h1.content )}</h1>`,
```
Comment (nataliashitova) [inline src/assessments/seo/SingleH1Assessment.js]: Fancy!

**Comment by nataliashitova** [comment]
Works like a charm in the testing environment. But unfortunately not in the plugin. 

The tag attributes are not correctly parsed, because tinyMCE changes single quotes to double quotes and the marking texts cannot be found back. This problem was previously discussed in https://github.com/Yoast/YoastSEO.js/issues/1897. @moorscode shall we still wait for the new HTML parser or shall we try to fix it now?

**Comment by moorscode** [comment]
As the tree structure is being developed currently, let's pick this up in that proces.

**Comment by bintzandt** [comment]
PR transferred to https://github.com/Yoast/javascript/pull/21

**Pull request closed without merging**